enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,192 @@
+---
+title: use-manifest-annotation-for-object-removal
+authors:
+  - "@jottofar"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@wking"
+  - "@smarterclayton"
+  - "@deads2k"
+approvers:
+  - "@smarterclayton"
+  - "@deads2k"
+creation-date: 2020-08-03
+last-updated: 2020-08-10
+status: implementable
+---
+
+# Use Manifest Annotation For Object Removal
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently CVO managed object removal is handled by jobs.
+Jobs have many immutable components, so they aren't a great match for CVO managed manifests.
+This enhancement replaces that approach with a new manifest annotation requesting the CVO delete the in-cluster object instead of creating/updating it.
+This will provide a more straightforward way for engineering to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Devlopers will be able to remove any of the currently managed CVO objects by modifying an existing manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for managment by CVO, therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+This enhancement neither adds nor requires any application specific delete logic within CVO.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO.
+Values other than `true` will result in CVO logging an error and no further processing of the manifest.
```

Comment:
This seems like potential a place for bugs to creep in at some point.  If an error is logged, is anyone alerted?  Should we have the CVO fail so this type of thing doesn't slip through CI?  "No further processing of the manifest" implies that if someone used the value of 'false' and intended to update other fields on their object, the new version of the object would not be applied because of this logic.

Reply:
Yeah,I'd rather this be fatal, CVOexits non-zero.  And possibly with a matching guard in `oc adm release new` to fail even sooner if someone typos the value.

Reply:
Made explicit mention of CVO CI test failure if value != `true` in commit [fb15be7](https://github.com/openshift/enhancements/pull/419/commits/fb15be7678555eeacff0c7eb630bbefdd4b0791b).